bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusCGateHandler.java

```diff
@@ -0,0 +1,542 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.cbus.handler;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.GregorianCalendar;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.cbus.CBusBindingConstants;
+
+import java.util.concurrent.Future;
+import java.util.concurrent.ThreadPoolExecutor;
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+
+import com.daveoxley.cbus.CGateConnectException;
+import com.daveoxley.cbus.CGateException;
+import com.daveoxley.cbus.CGateInterface;
+import com.daveoxley.cbus.CGateSession;
+import com.daveoxley.cbus.CGateThreadPool;
+import com.daveoxley.cbus.CGateThreadPoolExecutor;
+import com.daveoxley.cbus.events.EventCallback;
+import com.daveoxley.cbus.status.StatusChangeCallback;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+
+/**
+ * The {@link CBusThreadPool} is responsible for executing jobs from a threadpool
+ *
+ * @author John Harvey - Initial contribution
+ */
+
+@NonNullByDefault
+final class CBusThreadPool extends CGateThreadPool
```

Comment:
What is the purpose of this class? Why not just use the scheduler provided for ThingHandlers?

Reply:
This uses abstract intefaces within the cgate libraries so that it can use Openhab's threadpool instead of having its own. 
This was introduced following a comment from wborn (22Dec 207) in Pull request #2935 which was cancelled.

Reply:
I'll make note of that, thanks for letting me know.